lib/chef/resource/mount.rb

```diff
@@ -46,7 +46,8 @@ class Mount < Chef::Resource
                description: "The directory (or path) in which the device is to be mounted. Defaults to the name of the resource block if not provided."
 
       property :device, String, identity: true,
-               description: "Required for `:umount` and `:remount` actions (for the purpose of checking the mount command output for presence). The special block device or remote node, a label, or a uuid to be mounted."
+               description: "Required for `:umount` and `:remount` actions (for the purpose of checking the mount command output for presence). The special block device or remote node, a label, or a uuid to be mounted.",
+               coerce: proc { |x| x.include?(" ") ? x.gsub(" ", "\\x20") : x }
```

Comment:
this seems like the wrong change to make to the superclass that doesn't know anything about `findmnt` on linux.

it would probably be more correct to take the output of `findmnt` and convert the '\x20's back to spaces there, then ensure that the `shell_out` call to the `mount` command was using the array-style calling so that spaces were accepted as arguments.

otherwise you're injecting this behavior into every subclass and every call to mount to work around findmnt on unix.  this probably isn't appropriate on AIX or Solaris for example.

Reply:
Hi, We have made the changes but instead of substituting output of `findmnt` command line by line - '\x20` with space. We added one more when condition to check for spaces in device name. This will avoid the substitution happening for every line in findmnt output.